### ISSUE TITLE
Using model->predict(imageMatrix)

### ISSUE THREAD

**Issue by larivierec** [Itseez/opencv_contrib#435]
Good evening everyone,

I have tried to use this 

```c++
    int label = -1;
    double confidence = 0.0;
    this->imageManager = pManager;
    this->model = createEigenFaceRecognizer();
    this->model->train(pManager->getOpenCVTrainImages(), pManager->getTrainLabels());

    for(Mat& imageToTest : pManager->getOpenCVTestImages())
    {
        this->model->predict(imageToTest, label, confidence);
        cout << "Class: " << label
            << " The confidence predicted towards the test image is: " << confidence << endl;
    }
```
When running this code, it correctly identifies the label (which is a good thing) however, if the test image is run against a subset not part of its class I was expecting to receive a class of -1 or a really LOW confidence.

In this example I have not set the face recognizer to 10 PCA components and I did not set a threshold. However, I have tried to run it with a createEigenFaceRecognizer(10, 123.0) or createEigenFaceRecognizer(10, 50.0) or createEigenFaceRecognizer(0, 1.0) or even (80, 123.0)...

All of the above return me a confidence of 0. Which seems very unlikely or impossible I was expecting at least a few numbers.

What am I doing wrong, is this a bug?

**Comment by comdiv** [comment]
While you test your trained cards over themself - you will get  0 confidence because it means "copy" or "dublicate" - 0 is close as it can be. It's really high value. Value of confidence means DISTANCE from testing image.

If you want to test cross - correlation between images look for #402 (already merged) and #465 (pending) if you checkout yourself or wait merging - you can use `TopNPredictCollector` to get set or most closest (relevant) labels from set (not just one). Additionally you can disable including self label in result:
```
Mat myImageFromTrainSet = ...;
int myLabelOfTestingImage = ...;
Ptr<TopNPredictCollector> collector = TopNPredictCollector.create(5, THRESH); //create top 5 collector with threshold
collector.setExcludeLabel(myLabelOfTestingImage); // force collector ignore it's own label to not take itself with 0 confidence
model.predict( myImageFromTrainSet, collector ); //it will collect up to 5 most close labels excluding itself and it's brothers by train set
Ptr<std::list<std::pair<int,double>>> result = collector.getResult();
// do what you want - it's sorted and without not required things
```

**Comment by larivierec** [comment]
Wow, my apologies this was a huge misunderstanding on my part! Thank you so much
